dev-tools/omdb/src/bin/omdb/mgs/sensors.rs

```diff
@@ -0,0 +1,954 @@
+    pub verbose: bool,
+
+    /// restrict to specified sled(s)
+    #[clap(long, use_value_delimiter = true)]
+    pub sled: Vec<u32>,
+
+    /// exclude sleds rather than include them
+    #[clap(long, short)]
+    pub exclude: bool,
+
+    /// include switches
+    #[clap(long)]
+    pub switches: bool,
+
+    /// include PSC
+    #[clap(long)]
+    pub psc: bool,
+
+    /// print sensors every second
+    #[clap(long, short)]
+    pub sleep: bool,
+
+    /// parseable output
+    #[clap(long, short)]
+    pub parseable: bool,
+
+    /// show latencies
+    #[clap(long)]
+    pub show_latencies: bool,
+
+    /// restrict sensors by type of sensor
+    #[clap(
+        long,
+        short,
+        value_name = "sensor type",
+        use_value_delimiter = true
+    )]
+    pub types: Option<Vec<String>>,
+
+    /// restrict sensors by name
+    #[clap(
+        long,
+        short,
+        value_name = "sensor name",
+        use_value_delimiter = true
+    )]
+    pub named: Option<Vec<String>>,
+
+    /// simulate using specified file as input
+    #[clap(long, short)]
+    pub input: Option<String>,
+
+    /// start time, if using an input file
+    #[clap(long, value_name = "time", requires = "input")]
+    pub start: Option<u64>,
+
+    /// end time, if using an input file
+    #[clap(long, value_name = "time", requires = "input")]
+    pub end: Option<u64>,
+
+    /// duration, if using an input file
+    #[clap(
+        long,
+        value_name = "seconds",
+        requires = "input",
+        conflicts_with = "end"
+    )]
+    pub duration: Option<u64>,
+}
+
+impl SensorsArgs {
+    fn matches_sp(&self, sp: &SpIdentifier) -> bool {
+        match sp.type_ {
+            SpType::Sled => {
+                let matched = if self.sled.len() > 0 {
+                    self.sled.iter().any(|&v| v == sp.slot)
+                } else {
+                    true
+                };
+
+                matched != self.exclude
+            }
+            SpType::Switch => self.switches,
+            SpType::Power => self.psc,
+        }
+    }
+}
+
+#[derive(Clone, Debug, Hash, PartialEq, Eq, PartialOrd, Ord)]
+pub(crate) struct Sensor {
+    pub name: String,
+    pub kind: MeasurementKind,
+}
+
+impl Sensor {
+    fn units(&self) -> &str {
+        match self.kind {
+            MeasurementKind::Temperature => "°C",
+            MeasurementKind::Current | MeasurementKind::InputCurrent => "A",
+            MeasurementKind::Voltage | MeasurementKind::InputVoltage => "V",
+            MeasurementKind::Speed => "RPM",
+            MeasurementKind::Power => "W",
+        }
+    }
+
+    fn format(&self, value: f32, parseable: bool) -> String {
+        if parseable {
+            format!("{value}")
+        } else {
+            match self.kind {
+                MeasurementKind::Speed => {
+                    format!("{value:0} RPM")
+                }
+                _ => {
+                    format!("{value:.2}{}", self.units())
```

Comment:
Do you want a space between these like there is for `RPM`?

Reply:
Yeah: for the other units (`°C`, `V`, `A`), it looks more natural with the number next to the unit -- but it looks decidedly unnatural with `RPM`; will add a comment.